### ISSUE TITLE
[Android] Keyboard navigation does not work with RadioButtons with redefined appearance

### ISSUE THREAD

**Issue by kubaflo** [dotnet/maui#30040]
### Description

When using a RadioButton with a ControlTemplate, keyboard navigation (arrow keys or Tab) doesn't work. This behavior breaks accessibility and is inconsistent with standard RadioButton behavior.

The issue can be reproduced using the official sample from the MAUI documentation:
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-9.0#redefine-radiobutton-appearance

<video source="https://github.com/user-attachments/assets/a64ed444-f09f-4d1f-86de-98960b57b965"/>

### Steps to Reproduce

``` xaml
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="Maui.Controls.Sample.MainPage"
             xmlns:local="clr-namespace:Maui.Controls.Sample">
<ContentPage.Resources>
    <ControlTemplate x:Key="RadioButtonTemplate">
        <Border Stroke="#F3F2F1"
                StrokeThickness="2"
                StrokeShape="RoundRectangle 10"
                BackgroundColor="#F3F2F1"
                HeightRequest="90"
                WidthRequest="90"
                HorizontalOptions="Start"
                VerticalOptions="Start">
            <VisualStateManager.VisualStateGroups>
                <VisualStateGroupList>
                    <VisualStateGroup x:Name="CheckedStates">
                        <VisualState x:Name="Checked">
                            <VisualState.Setters>
                                <Setter Property="Stroke"
                                        Value="#FF3300"/>
                                <Setter TargetName="check"
                                        Property="Opacity"
                                        Value="1"/>
                            </VisualState.Setters>
                        </VisualState>
                        <VisualState x:Name="Unchecked">
                            <VisualState.Setters>
                                <Setter Property="BackgroundColor"
                                        Value="#F3F2F1"/>
                                <Setter Property="Stroke"
                                        Value="#F3F2F1"/>
                                <Setter TargetName="check"
                                        Property="Opacity"
                                        Value="0"/>
                            </VisualState.Setters>
                        </VisualState>
                    </VisualStateGroup>
                </VisualStateGroupList>
            </VisualStateManager.VisualStateGroups>
            <Grid Margin="4"
                    WidthRequest="90">
                <Grid Margin="0,0,4,0"
                        WidthRequest="18"
                        HeightRequest="18"
                        HorizontalOptions="End"
                        VerticalOptions="Start">
                    <Ellipse Stroke="Blue"
                                Fill="White"
                                WidthRequest="16"
                                HeightRequest="16"
                                HorizontalOptions="Center"
                                VerticalOptions="Center"/>
                    <Ellipse x:Name="check"
                                Fill="Blue"
                                WidthRequest="8"
                                HeightRequest="8"
                                HorizontalOptions="Center"
                                VerticalOptions="Center"/>
                </Grid>
                <ContentPresenter/>
            </Grid>
        </Border>
    </ControlTemplate>
</ContentPage.Resources>

<StackLayout>
    <Label Text="What's your favorite animal?"/>
    <StackLayout RadioButtonGroup.GroupName="animals"
                    Orientation="Horizontal">
        <StackLayout.Resources>
            <Style TargetType="RadioButton">
                <Setter Property="ControlTemplate"
                        Value="{StaticResource RadioButtonTemplate}"/>
            </Style>
        </StackLayout.Resources>
        <RadioButton Value="Cat">
            <RadioButton.Content>
                <StackLayout>
                    <Image Source="dotnet_bot.png"
                            HeightRequest="50"
                            HorizontalOptions="Center"
                            VerticalOptions="Center"/>
                    <Label Text="Cat"
                            HorizontalOptions="Center"
                            VerticalOptions="End"/>
                </StackLayout>
            </RadioButton.Content>
        </RadioButton>
        <RadioButton Value="Dog">
            <RadioButton.Content>
                <StackLayout>
                    <Image Source="dotnet_bot.png"
                            HeightRequest="50"
                            HorizontalOptions="Center"
                            VerticalOptions="Center"/>
                    <Label Text="Dog"
                            HorizontalOptions="Center"
                            VerticalOptions="End"/>
                </StackLayout>
            </RadioButton.Content>
        </RadioButton>
        <RadioButton Value="Elephant">
            <RadioButton.Content>
                <StackLayout>
                    <Image Source="dotnet_bot.png"
                            HeightRequest="50"
                            HorizontalOptions="Center"
                            VerticalOptions="Center"/>
                    <Label Text="Elephant"
                            HorizontalOptions="Center"
                            VerticalOptions="End"/>
                </StackLayout>
            </RadioButton.Content>
        </RadioButton>
        <RadioButton Value="Monkey">
            <RadioButton.Content>
                <StackLayout>
                    <Image Source="dotnet_bot.png"
                            HeightRequest="50"
                            HorizontalOptions="Center"
                            VerticalOptions="Center"/>
                    <Label Text="Monkey"
                            HorizontalOptions="Center"
                            VerticalOptions="End"/>
                </StackLayout>
            </RadioButton.Content>
        </RadioButton>
    </StackLayout>

    <VerticalStackLayout  RadioButtonGroup.GroupName="animals2">
        <RadioButton GroupName="animals2"
                        Value="Cat2"
                        Content="Cat"/>
        <RadioButton GroupName="animals2"
                        Value="Dog2"
                        Content="Dog"/>
        <RadioButton GroupName="animals2"
                        Value="Elephant2"
                        Content="Elephant"/>
        <RadioButton GroupName="animals2"
                        Value="Monkey2"
                        Content="Monkey"/>
    </VerticalStackLayout>
</StackLayout>
</ContentPage>
```

### Version with bug

9.0.70 SR7

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Relevant log output

```shell

```
